pkg/image/resolve.go

```diff
@@ -119,20 +119,37 @@ func generateImages(chartNameAndVersion string, inputMap map[interface{}]interfa
 		return
 	}
 
-	// distinguish images by os
```

Comment:
https://github.com/rancher/rancher/blob/f6dfe7ca192c4f87c4d197ab95888a1bf15d6dab/pkg/image/resolve.go#L168-L170

Can we add another if statement here that says:

```go
 if v, ok := value.([]interface{}); ok {
    for _, elem := range v {
        walkthroughMap(v, walkFunc) 
    }
 }
```

I can make this in a separate PR, but it's a really small change. This currently still breaks wins-upgrader, since the image exists within a list as seen here:

https://github.com/rancher/charts/blob/99131a36a40414789a9f3de3930bf7988a23a85e/charts/rancher-wins-upgrader/rancher-wins-upgrader/0.0.100-rc01/values.yaml#L19-L22

Reply:
I'm fine with this change so long as @cbron is. However, this would allow for more options to our charts' images format.

Reply:
Will move into another PR

Reply:
https://github.com/rancher/rancher/pull/32324